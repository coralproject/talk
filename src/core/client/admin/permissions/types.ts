import { GQLUSER_ROLE_RL } from "coral-framework/schema";

export type PermissionMap<AbilityType extends string, ContextType> = {
  [abilityType in AbilityType]: {
    [role in GQLUSER_ROLE_RL]?: (context?: ContextType) => boolean;
  };
};
