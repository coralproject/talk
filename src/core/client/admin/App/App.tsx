import { BrowserProtocol, queryMiddleware } from "farce";
import { createFarceRouter, ElementsRenderer } from "found";
import { Resolver } from "found-relay";
import React, { FunctionComponent } from "react";

import { CoralContextConsumer } from "coral-framework/lib/bootstrap/CoralContext";
import TransitionControl from "coral-framework/testHelpers/TransitionControl";

import routeConfig from "../routeConfig";
import NotFound from "../routes/NotFound";

const Router = createFarceRouter({
  historyProtocol: new BrowserProtocol(),
  historyMiddlewares: [queryMiddleware],
  routeConfig,
  renderReady: function FarceRouterReady({ elements }) {
    return (
      <>
        <ElementsRenderer elements={elements} />
        {process.env.NODE_ENV === "test" && <TransitionControl />}
      </>
    );
  },
  renderError: function FarceRouterError({ error }) {
    return <div>{error.status === 404 ? <NotFound /> : "Error"}</div>;
  },
});

const App: FunctionComponent = () => (
  <CoralContextConsumer>
    {({ relayEnvironment }) => (
      <Router resolver={new Resolver(relayEnvironment)} />
    )}
  </CoralContextConsumer>
);

export default App;
