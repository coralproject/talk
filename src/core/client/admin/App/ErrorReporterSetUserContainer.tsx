import { FunctionComponent, useEffect } from "react";
import { graphql } from "react-relay";

import { useEffectAtUnmount } from "coral-framework/hooks";
import { globalErrorReporter } from "coral-framework/lib/errors";
import { withFragmentContainer } from "coral-framework/lib/relay";

import { ErrorReporterSetUserContainer_viewer } from "coral-stream/__generated__/ErrorReporterSetUserContainer_viewer.graphql";

interface Props {
  viewer: ErrorReporterSetUserContainer_viewer | null;
}

const ErrorReporterSetUserContainer: FunctionComponent<Props> = ({
  viewer,
}) => {
  useEffect(() => {
    globalErrorReporter.setUser(viewer);
  }, [viewer]);

  useEffectAtUnmount(() => {
    globalErrorReporter.setUser(null);
  });
  return null;
};

const enhanced = withFragmentContainer<Props>({
  viewer: graphql`
    fragment ErrorReporterSetUserContainer_viewer on User {
      id
      username
      role
    }
  `,
})(ErrorReporterSetUserContainer);

export default enhanced;
