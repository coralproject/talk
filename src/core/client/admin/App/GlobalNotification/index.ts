export { NotificationProvider } from "./GlobalNotificationContext";
export { default as NotificationContainer } from "./NotificationContainer";
export { default as useNotification } from "./useNotification";
