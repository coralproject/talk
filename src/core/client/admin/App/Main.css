:global {
  body {
    padding: 0;
    margin: 0;
  }
}

.root {
}

.logoContainer {
  position: relative;
}
