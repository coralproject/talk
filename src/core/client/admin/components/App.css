:global {
  body {
    padding: 0;
    margin: 0;
  }
}

.container {
  max-width: 1280px;
  margin: 0 auto;
  padding: 0 var(--spacing-unit);
  box-sizing: border-box;
}
