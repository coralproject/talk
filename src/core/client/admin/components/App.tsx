import React, { StatelessComponent } from "react";

import styles from "./App.css";
import AppBar from "./AppBar";
import Navigation from "./Navigation";

const App: StatelessComponent = ({ children }) => (
  <div>
    <AppBar>
      <Navigation />
    </AppBar>
    <div className={styles.container}>{children}</div>
  </div>
);

export default App;
