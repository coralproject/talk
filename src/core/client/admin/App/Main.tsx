import React, { FunctionComponent } from "react";

import { PropTypesOf } from "coral-framework/types";
import { LogoHorizontal } from "coral-ui/components";
import { AppBar, Begin, Divider, End } from "coral-ui/components/v2/AppBar";

import { DecisionHistoryButton } from "./DecisionHistory";
import {
  NotificationContainer,
  NotificationProvider,
} from "./GlobalNotification";
import NavigationContainer from "./Navigation";
import UserMenuContainer from "./UserMenu";
import Version from "./Version";

import styles from "./Main.css";

interface Props {
  viewer: PropTypesOf<typeof UserMenuContainer>["viewer"] &
    PropTypesOf<typeof NavigationContainer>["viewer"];
  children: React.ReactNode;
}

const Main: FunctionComponent<Props> = ({ children, viewer }) => (
  <div className={styles.root}>
    <AppBar gutterBegin gutterEnd>
      <Begin itemGutter="double">
        <div className={styles.logoContainer}>
          <LogoHorizontal />
        </div>
        <NavigationContainer viewer={viewer} />
      </Begin>
      <End>
        <DecisionHistoryButton />
        <Divider />
        <UserMenuContainer viewer={viewer} />
      </End>
    </AppBar>
    <NotificationProvider>
      <NotificationContainer />
      {children}
    </NotificationProvider>
    <Version />
  </div>
);

export default Main;
