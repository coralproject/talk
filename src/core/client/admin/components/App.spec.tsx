import { shallow } from "enzyme";
import React from "react";

import { PropTypesOf } from "talk-framework/types";

import App from "./App";

it("renders correctly", () => {
  const props: PropTypesOf<typeof App> = {
    children: "child",
  };
  const wrapper = shallow(<App {...props} />);
  expect(wrapper).toMatchSnapshot();
});
