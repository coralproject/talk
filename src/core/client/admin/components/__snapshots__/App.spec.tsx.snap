// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <AppBar>
    <Navigation />
  </AppBar>
  <div
    className="App-container"
  >
    child
  </div>
</div>
`;
